src/CMakeLists.txt

```diff
@@ -262,32 +260,72 @@ if(BUILD_STATIC)
     endif()
 endif()
 
-if (ENABLE_LTO)
-    # Exports compilation unit is entirely a hack. It references
-    # symbols among static libraries and object files declaring
-    # them all as functions. To avoid header dependencies. This is
-    # not detected by the compilers, since they never see
-    # conflicting definitions in one compilation unit. But this
-    # makes LTO mad, because the linker sees all the definitions,
-    # and is especially angry when a variable is declared as a
-    # function. To get rid of these false positive errors the
-    # exports file is not link-time optimized.
-    set_source_files_properties(exports.c PROPERTIES COMPILE_FLAGS -fno-lto)
+set(exports_file_sources ${CMAKE_SOURCE_DIR}/extra/exports
+    ${CMAKE_SOURCE_DIR}/extra/exports_xxhash)
+if (EXPORT_LIBCURL_SYMBOLS)
+    set(exports_file_sources ${exports_file_sources}
+        ${CMAKE_SOURCE_DIR}/extra/exports_libcurl)
 endif()
+string(REPLACE ";" " " exports_file_sources "${exports_file_sources}")
+
+# Exports syntax is toolchain-dependent, preprocessing is necessary
+set(exports_file ${CMAKE_BINARY_DIR}/extra/exports.${CMAKE_SYSTEM_NAME})
+add_custom_target(preprocess_exports
+                  DEPENDS ${exports_file})
+add_custom_command(
+    OUTPUT  ${exports_file}
+    DEPENDS ${CMAKE_SOURCE_DIR}/extra/exports
```

Comment:
It should also depend on exports_xxhash since you have it separated. Or better - depend on `exports_file_sources`, otherwise it won't depend on curl exports sometimes.

Reply:
I missed the point: why we ever need to depend on always existing source files?

Reply:
What if you change it locally by adding some new symbols? Shouldn't the stuff be rebuilt then? I thought DEPENDS is about that, isn't it?

Reply:
Sure, my bad. Thanks.

Reply:
Update "DEPENDS".